rust/libnewsboat/src/logger.rs

```diff
@@ -99,15 +117,11 @@ impl fmt::Display for Level {
 /// logger.log(Level::Debug, &format!("feeds.len() == {}", 42));
 /// ```
 pub struct Logger {
-    /// The file to which all messages at and above `loglevel` will be written.
-    logfile: Option<File>,
-
-    /// The file to which all Level::UserError messages will be written if `loglevel` is not
-    /// Level::None.
-    errorlogfile: Option<File>,
+    /// Handles for the files to which messages should be written.
+    files: Mutex<LogFiles>,
```

Comment:
Is locking LogFiles behind a mutex necessary? I may be reading the Rust Docs wrong but it looks like File types implement `Send` meaning they are thread safe. Since Both of LogFiles members are `Send` that means LogFile is as well, right?

Reply:
Yep, `LogFiles` is `Send`, but that trait doesn't guarantee thread safety—it only lets us move the value to another thread. We could probably design a logger that's passed from thread to thread (like a token), but that would still require a queue of some sort (which will introduce synchronization).

`File`, and thus `LogFiles`, is also `Sync`, which means we can share `&LogFiles` between threads without synchronization. This is not enough for our use case, since we need `&mut File` to write to the log. That's why we need `Mutex` here.

Thank you for taking the time to read and understand the PR, it's very appreciated!